From 7c6c2cdf9e8a060eec7596abc1c85a094d908757 Mon Sep 17 00:00:00 2001
From: Ken Tseng <ken.tseng@realtek.com>
Date: Tue, 8 Mar 2022 14:34:59 +0800
Subject: [PATCH 43/54] 824: spi: Add SPI_DW_RTK


- Add spi nor and nand flash controller driver for kent
---
 drivers/spi/Kconfig  | 9 +++++++++
 drivers/spi/Makefile | 1 +
 2 files changed, 10 insertions(+)

diff --git a/drivers/spi/Kconfig b/drivers/spi/Kconfig
index aadaea052..1eb6cc80e 100644
--- a/drivers/spi/Kconfig
+++ b/drivers/spi/Kconfig
@@ -282,6 +282,15 @@ config SPI_DW_BT1_DIRMAP
 
 endif
 
+config SPI_DW_RTK
+	tristate "Realtek SPI controller based on DW SPI core"
+	select SPI_DESIGNWARE
+	depends on ARCH_REALTEK
+	default n
+	help
+	  Driver for the SPI controller core from DesignWare on
+	  Realtek RTD139x, RTD16xx, RTD13xx, or RTD16xxb platform.
+
 config SPI_DLN2
        tristate "Diolan DLN-2 USB SPI adapter"
        depends on MFD_DLN2
@@ -940,6 +940,19 @@ config SPI_UNIPHIER
 
 	  If your SoC supports SCSSI, say Y here.
 
+config SPI_RTKNOR
+	tristate "SPI device driver support for RTK NOR"
+	depends on SPI_MASTER
+	help
+	  Enables SPI Nor controller driver for RTK SoC.
+
+config SPI_RTK_NAND
+	tristate "RTK SPI NAND controller"
+	depends on SPI_MASTER
+	select MTD_NAND_ECC
+	help
+	  Enables SPI Nand controller driver for RTK SoC.
+
 config SPI_XCOMM
 	tristate "Analog Devices AD-FMCOMMS1-EBZ SPI-I2C-bridge driver"
 	depends on I2C
diff --git a/drivers/spi/Makefile b/drivers/spi/Makefile
index 6fea58216..a37cab5df 100644
--- a/drivers/spi/Makefile
+++ b/drivers/spi/Makefile
@@ -42,6 +42,9 @@ spi-dw-$(CONFIG_SPI_DW_DMA)		+= spi-dw-dma.o
 obj-$(CONFIG_SPI_DW_BT1)		+= spi-dw-bt1.o
 obj-$(CONFIG_SPI_DW_MMIO)		+= spi-dw-mmio.o
 obj-$(CONFIG_SPI_DW_PCI)		+= spi-dw-pci.o
+obj-$(CONFIG_SPI_DW_RTK)		+= spi-dw-rtk.o
+obj-$(CONFIG_SPI_RTKNOR)                += spi-rtknor.o
+obj-$(CONFIG_SPI_RTK_NAND)              += spi-rtk-sn.o
 obj-$(CONFIG_SPI_EFM32)			+= spi-efm32.o
 obj-$(CONFIG_SPI_EP93XX)		+= spi-ep93xx.o
 obj-$(CONFIG_SPI_FALCON)		+= spi-falcon.o
-- 
2.17.1

